### PR TITLE
Update Unit method getParentUnitsPresentationName

### PR DESCRIPTION
**Pull request by pedrovpmadeira**

### CODE REVIEW

File: src/test/java/org/fenixedu/academic/domain/UnitTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package org.fenixedu.academic.domain;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (Sofia-Nascimento) [inline src/test/java/org/fenixedu/academic/domain/UnitTest.java:3]: We usually use: org.junit.Assert.assertTrue

File: src/test/java/org/fenixedu/academic/domain/UnitTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        uType.addAllowedParentConnectionRules(new ConnectionRule(pType, null, aType));
+
+        testParent = Unit.createNewUnit(Optional.ofNullable(pType), new LocalizedString(Locale.ENGLISH, "PARENT_UNIT"), "PARENT",
+                null, aType);
+        testUnit = Unit.createNewUnit(Optional.ofNullable(uType), new LocalizedString(Locale.ENGLISH, "NEW_UNIT"), "NEW",
+                testParent, aType);
+    }
+
+    @Test
+    public void testGetParentUnitsPresentationNameSucessful() {
```
Comment (Sofia-Nascimento) [inline src/test/java/org/fenixedu/academic/domain/UnitTest.java:55]: We usually write our test methods with test<NameOfClass>_<caseBeingTested>

In this case we're testing the Units, if the presentation name comes in the format expected. 
Perhaps something like this? `testUnits_presentationName`

File: src/test/java/org/fenixedu/academic/domain/UnitTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.fenixedu.commons.i18n.LocalizedString;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.FenixFrameworkRunner;
+
+import pt.ist.fenixframework.FenixFramework;
+
+@RunWith(FenixFrameworkRunner.class)
+public class UnitTest {
```
Comment (Sofia-Nascimento) [inline src/test/java/org/fenixedu/academic/domain/UnitTest.java:23]: There's already a test class for Units, called OrganizationalStructureTest. I believe this test fits there best.

File: src/test/java/org/fenixedu/academic/domain/UnitTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                null, aType);
+        testUnit = Unit.createNewUnit(Optional.ofNullable(uType), new LocalizedString(Locale.ENGLISH, "NEW_UNIT"), "NEW",
+                testParent, aType);
+    }
+
+    @Test
+    public void testGetParentUnitsPresentationNameSucessful() {
+        testUnit.getAllSubUnits().stream().forEach(unit -> {
+            assertTrue(StringUtils.countMatches(unit.getParentUnitsPresentationName(" > "),
+                    ">") == unit.getParentUnitsPath().stream().filter(parent -> !parent.isAggregateUnit()).count() - 1);
```
Comment (Sofia-Nascimento) [inline src/test/java/org/fenixedu/academic/domain/UnitTest.java:58]: I'm not sure this is a good test. According to your variables, this is the child of the testParent. So it doesn't have subUnits, and **therefore the forEach is not running.** However, let's assume you do `testParent.getAllSubUnits()`: you're only checking that the subunit (testUnit) has one parent, not **the** parent. 

You can verify the count first, but then I would compare with a String (made by hand) or what's expected of the presentation name for that Unit. Test both child and parent.

- [ ] Make the test bigger (you want **a** path to appear)
- [ ] Compare the results with a presentation name made by hand
- [ ] Test with Aggregate units in the path

**Review by sanavarali — APPROVED** [review]

**Review by Sofia-Nascimento — APPROVED** [review]

**Pull request merged**